crates/cxx-qt-gen/src/parser/property.rs

```diff
@@ -3,14 +3,112 @@
 //
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
-use syn::{parse::ParseStream, Attribute, Ident, Result, Token, Type};
+use syn::{
+    parse::{Error, ParseStream},
+    punctuated::Punctuated,
+    spanned::Spanned,
+    Attribute, Expr, Ident, Meta, MetaNameValue, Path, Result, Token, Type,
+};
+
+/// An optional identifier of the functions passed with flags
+/// e.g. read = my_getter, IdentFlag would be used to store my_getter
+type IdentFlag = Option<Ident>;
+
+/// Struct for storing the flags provided for a QProperty and their optional identifiers ([IdentFlag])
+#[derive(Debug)]
+pub struct QPropertyFlags {
+    // TODO: Refactor to use an enum type for representing all 3 states of write and notify
+    // Or use an enum for {Auto, Custom(Ident)}, this is optional for write and notify, but not read
+    pub(crate) read: IdentFlag,
+    pub(crate) write: Option<IdentFlag>,
+    pub(crate) notify: Option<IdentFlag>,
+}
+
+impl Default for QPropertyFlags {
+    // Default represents the flags of the desugared version of #[qproperty(T, ident)]
+    fn default() -> Self {
+        Self {
+            read: None,
+            write: Some(None),
+            notify: Some(None),
+        }
+    }
+}
+
+impl QPropertyFlags {
+    // Doesn't really represent a realistic state, just used for collecting in the parser
+    pub fn new() -> Self {
```

Comment:
as this is only (?) used in this file and only once (?) maybe instead of making it a `pub` method either keep it private or at least `pub (crate)` or just use `QPropertyFlags { read: None, write: None, notify: None }` where the new() is called?

Or instead build them up before creating it so that you don't need read to be None

Reply:
Read was only one Option, so None means it exists, just has no Identifier. This should also be fixed with the move to a State enum inside QPropertyFlags

Reply:
done

Reply:
this is unused now? think clippy was picking that up

Reply:
done